parachain/pallets/bridge/src/lib.rs

```diff
@@ -65,28 +110,32 @@ decl_module! {
 
 		fn deposit_event() = default;
 
-		/// Submit `message` for dispatch to a target application identified by `app_id`.
 		#[weight = 0]
-		pub fn submit(origin, app_id: AppId, message: Message) -> DispatchResult {
+		pub fn submit(origin, channel_id: ChannelId, message: Message) -> DispatchResult {
 			let who = ensure_signed(origin)?;
 
-			// TODO: move replay protection here
+			// Construct channel object from storage
+			let channel = Self::get_inbound_channel(channel_id);
+
+			// Design choices:
+			//   Do verification and then submit to channel?
```

Comment:
there needs to be some verification inside the channel, for example, the receiving channel needs to verify the smart contract address of the sending channel to make sure it comes from a trusted source. i think maybe all the verification could go in the channel even?